ArchiSteamFarm/Steam/Interaction/Commands.cs

```diff
@@ -926,6 +999,121 @@ internal void OnNewLicenseList() {
 		return responses.Count > 0 ? string.Join(Environment.NewLine, responses) : null;
 	}
 
+	private async Task<string?> ResponseAdvancedTransferByAssetRarity(EAccess access, uint appID, ulong contextID, Bot targetBot, HashSet<EAssetRarity> assetRarities) {
+		if (!Enum.IsDefined(access)) {
+			throw new InvalidEnumArgumentException(nameof(access), (int) access, typeof(EAccess));
+		}
+
+		ArgumentOutOfRangeException.ThrowIfZero(appID);
+		ArgumentOutOfRangeException.ThrowIfZero(contextID);
+		ArgumentNullException.ThrowIfNull(targetBot);
+
+		if (access < EAccess.Master) {
+			return null;
+		}
+
+		if (!Bot.IsConnectedAndLoggedOn) {
+			return FormatBotResponse(Strings.BotNotConnected);
+		}
+
+		if (!targetBot.IsConnectedAndLoggedOn) {
+			return FormatBotResponse(Strings.TargetBotNotConnected);
+		}
+
+		(bool success, string message) = await Bot.Actions.SendInventory(appID, contextID, targetBot.SteamID, filterFunction: item => assetRarities.Contains(item.Rarity)).ConfigureAwait(false);
+
+		return FormatBotResponse(success ? message : Strings.FormatWarningFailedWithError(message));
+	}
+
+	private async Task<string?> ResponseAdvancedTransferByAssetRarity(EAccess access, string targetAppID, string targetContextID, string botNameTo, string assetRaritiesText) {
+		if (!Enum.IsDefined(access)) {
+			throw new InvalidEnumArgumentException(nameof(access), (int) access, typeof(EAccess));
+		}
+
+		ArgumentException.ThrowIfNullOrEmpty(targetAppID);
+		ArgumentException.ThrowIfNullOrEmpty(targetContextID);
+		ArgumentException.ThrowIfNullOrEmpty(botNameTo);
+		ArgumentException.ThrowIfNullOrEmpty(assetRaritiesText);
+
+		Bot? targetBot = Bot.GetBot(botNameTo);
+
+		if (targetBot == null) {
+			return access >= EAccess.Owner ? FormatBotResponse(Strings.FormatBotNotFound(botNameTo)) : null;
+		}
+
+		if (!uint.TryParse(targetAppID, out uint appID) || (appID == 0)) {
+			return FormatBotResponse(Strings.FormatErrorIsInvalid(nameof(appID)));
+		}
+
+		if (!ulong.TryParse(targetContextID, out ulong contextID) || (contextID == 0)) {
+			return FormatBotResponse(Strings.FormatErrorIsInvalid(nameof(contextID)));
+		}
+
+		string[] assetRaritiesArgs = assetRaritiesText.Split(SharedInfo.ListElementSeparators, StringSplitOptions.RemoveEmptyEntries);
+
+		HashSet<EAssetRarity> assetRarities = [];
+
+		foreach (string assetRarityArg in assetRaritiesArgs) {
+			if (!Enum.TryParse(assetRarityArg, true, out EAssetRarity assetRarity) || !Enum.IsDefined(assetRarity)) {
+				return FormatStaticResponse(Strings.FormatErrorIsInvalid(nameof(assetRarity)));
+			}
+
+			assetRarities.Add(assetRarity);
+		}
```

Comment:
Logic is same as above and could be extracted to method `HashSet<EAssetRarity> ParseAssetRarieties(string arg)`

Reply:
It's not general utility and it's exclusively used for parsing commands input so bottom of commands is appropriate.